[AIRFLOW-XXX] Update Contributing Guide - Git Hooks

feluelle opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
I thought of how you can help reducing failures in the Travis CI Pipeline and I came up with the idea of introducing Git Hooks. Read the changes for more information.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
There are no tests since it is only a change to the docs.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

feluelle committed:
[AIRFLOW-XXX] Update CONTRIBUTING - Git Hooks

ashb commented on CONTRIBUTING.md:
````diff
@@ -183,6 +183,24 @@ docker-compose -f scripts/ci/docker-compose.yml run airflow-testing /app/scripts
+
+NOTE: Change the `airflow-py2` and `airflow-py3` to your docker containers or remove the `docker exec` if you have set up your environment directly on your host system.
+```
+git diff upstream/master -u -- "*.py" | flake8 --diff
````
This chould just be `flake8` now - we shouldn't have any flake8 failures anymore, and checking the entire code base won't take more than a few seconds(?)

feluelle replied:
Thats true. I will change this.

feluelle commented on CONTRIBUTING.md:
````diff
@@ -183,6 +183,24 @@ docker-compose -f scripts/ci/docker-compose.yml run airflow-testing /app/scripts
+- test your code in a docker container based on python 3
+
+NOTE: Change the `airflow-py2` and `airflow-py3` to your docker containers or remove the `docker exec` if you have set up your environment directly on your host system.
+```
````
I am going to add `|| exit "$?"` after each command because it is necessary to return the exit code from the command to the shell script. So it exits if one of the commands return a non-zero exit code.

ashb replied:
Could also add `set -e` at the start (which I find cleaner than having `|| exit`  on each command personally.)

feluelle replied:
Didn't know that existed. Thank you. I will also change this.

feluelle committed:
[AIRFLOW-XXX] Refactor pre-commit example

- changes pre-commit example to use methods
- adds activating virtual env for python to run things like flake8 locally
- changes pre-commit file to use set -e command to instantly exit if any non-zero error occurs
- changes flake8 call to lint the repo instead of not only the changes files

feluelle committed:
[AIRFLOW-XXX] More refactoring of pre-commit
